lib/transport/transport-tls.c

```diff
@@ -135,8 +135,11 @@ log_transport_tls_read_method(LogTransport *s, gpointer buf, gsize buflen, LogTr
               rc = (log_transport_tls_send_shutdown(self) >= 0) ? 0 : -1;
               break;
             case SSL_ERROR_SYSCALL:
-              rc = -1;
-              /* errno is set accordingly */
+              // https://github.com/openssl/openssl/pull/10907
```

Comment:
I think referencing #11400 here would be more direct, because that PR contains the reasoning about the revert and proof that SSL_ERROR_SYSCALL+errno=0 indicates an unexpected EOF.

Reply:
Thank You! Also updated the PR description.